src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs

```diff
@@ -13,8 +14,15 @@ public partial class CompositionSpriteShape : CompositionShape
 		private SKPaint? _strokePaint;
 		private SKPaint? _fillPaint;
 
+		internal SKPath? LastDrawnStrokePath { get; private set; }
+		internal SKPath? LastDrawnFillPath { get; private set; }
```

Comment:
Not sure if there is an unmanaged memory leak here.

Reply:
Yeah, I noticed that too. According to the documentation, the unmanaged memory will be freed in the managed object's destructor.

Reply:
Disposing sometimes throws weird native errors. It should be safe to not dispose. The remark on `Dispose` says this:
>Always dispose the object before you release your last reference to the <see cref="T:SkiaSharp.SKNativeObject" />. Otherwise, the resources it is using will not be freed until the garbage collector calls the finalizer.

![image](https://github.com/unoplatform/uno/assets/66218781/20d29002-45a9-45df-93fe-5f78238ef4a6)

Reply:
Not disposing and relying on the GC may have performance impacts, I think. Finalizers aren't cheap, and an explicit Dispose call should be calling `GC.SuppressFinalize` (inside SkiaSharp)

What errors did you get when disposing?